### PR TITLE
ruff: add pyupgrade

### PR DESCRIPTION
**Pull request by ikonst**
For example, this [review comment](https://github.com/python/mypy/pull/15481#discussion_r1313755961) could've been spared with [UP036](https://beta.ruff.rs/docs/rules/outdated-version-block/).

### CODE REVIEW

**Comment by ikonst** [comment]
Turns out the [one UP037 exemption](https://github.com/python/mypy/pull/16023/commits/8d73ccd6a56b98566389facbf6b91002583003ee) already came up before:
https://github.com/python/mypy/pull/15575#discussion_r1251144634

**Comment by ichard26** [comment]
> @ichard26 is there an existing issue for this?
There might be an issue somewhere, but honestly it seems fine to include in https://github.com/mypyc/mypyc/issues/988.

**Review by hauntsaninja — APPROVED** [review]
Thanks!

File: mypy/config_parser.py [hunk 1]
```diff
@@ -329,7 +334,7 @@ def get_prefix(file_read: str, name: str) -> str:
     else:
         module_name_str = name
 
-    return f"{file_read}: [{module_name_str}]: "
```
Comment (AlexWaygood) [inline mypy/config_parser.py:332]: What's the purpose of the change to this line?

Comment (ikonst) [reply mypy/config_parser.py:332]: Before it was `print("%sfoobar" % prefix)`, now it's `print(prefix, "foobar")` hence removed the space.
